tests/integration_tests/test_collection_hybrid_search.py

```
@@ -0,0 +1,431 @@
+            if dimension <= len(base_vec):
+                embedding = base_vec[:dimension]
+            else:
+                embedding = base_vec * ((dimension // len(base_vec)) + 1)
+                embedding = embedding[:dimension]
+            
+            vector_str = "[" + ",".join(map(str, embedding)) + "]"
+            metadata_str = json.dumps(data["metadata"], ensure_ascii=False).replace("'", "\\'")
+            document_str = data["document"].replace("'", "\\'")
+            
+            sql = f"""INSERT INTO `{table_name}` (_id, document, embedding, metadata) 
+                     VALUES (CAST('{id_str_escaped}' AS BINARY), '{document_str}', '{vector_str}', '{metadata_str}')"""
```

Ruff (0.14.8)

Possible SQL injection vector through string-based query construction

(S608)